File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.cpp
class Solution {
public:
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        unordered_set<char> s = {'a', 'e', 'i', 'o', 'u'};
        int n = words.size();
        vector<int> pre(n + 1, 0);

        for (int i = 0; i < n; i++) {
            int cnt = (s.count(words[i].front()) && s.count(words[i].back())) ? 1 : 0;
            pre[i + 1] = pre[i] + cnt;
        }

        vector<int> res;
        for (auto& pair : queries) {
            res.push_back(pre[pair[1] + 1] - pre[pair[0]]);
        }

        return res;
    }
};

File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, w: List[str], q: List[List[int]]) -> List[int]:
        n = len(w)
        pre = [0] * (n + 1)          # [0, 1, 1, 2, 3, 4]
        s = "aeiou"

        v = 0

        for i in range(n):
            v = 0
            
            if w[i][0] in s and w[i][-1] in s:
                v = 1

            pre[i + 1] = pre[i] + v

        res = []
        for l,r in q:
            res.append(pre[r+1] - pre[l])

        return res